[gamenetworkingsockets] target_link_libraries failure

**Host Environment**

- OS: Arch Linux x64 (x64-linux)
- Compiler: clang 14.0.6

**To Reproduce**

Steps to reproduce the behavior:

`./vcpkg install gamenetworkingsockets`

use this in CMakeLists.txt

```
find_package(GameNetworkingSockets CONFIG REQUIRED)
target_link_libraries(test PRIVATE GameNetworkingSockets::shared GameNetworkingSockets::static GameNetworkingSockets::GameNetworkingSockets GameNetworkingSockets::GameNetworkingSockets_s)
```

**Failure logs**

```
-- Running vcpkg install
Detecting compiler hash for triplet x64-linux...
All requested packages are currently installed.
Restored 0 package(s) from /home/koriss/.cache/vcpkg/archives in 1.16 us. Use --debug to see more details.
Total install time: 8.56 us
gamenetworkingsockets provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(GameNetworkingSockets CONFIG REQUIRED)
    target_link_libraries(main PRIVATE GameNetworkingSockets::shared GameNetworkingSockets::static GameNetworkingSockets::GameNetworkingSockets GameNetworkingSockets::GameNetworkingSockets_s)

-- Running vcpkg install - done
-- Configuring done
CMake Error at build/vcpkg_installed/x64-linux/share/gamenetworkingsockets/GameNetworkingSockets.cmake:60 (set_target_properties):
  The link interface of target "GameNetworkingSockets::GameNetworkingSockets"
  contains:

    protobuf::libprotobuf

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  build/vcpkg_installed/x64-linux/share/gamenetworkingsockets/GameNetworkingSocketsConfig.cmake:47 (include)
  vcpkg/scripts/buildsystems/vcpkg.cmake:843 (_find_package)
  CMakeLists.txt:12 (find_package)


CMake Error at build/vcpkg_installed/x64-linux/share/gamenetworkingsockets/GameNetworkingSockets.cmake:70 (set_target_properties):
  The link interface of target
  "GameNetworkingSockets::GameNetworkingSockets_s" contains:

    protobuf::libprotobuf

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  build/vcpkg_installed/x64-linux/share/gamenetworkingsockets/GameNetworkingSocketsConfig.cmake:47 (include)
  vcpkg/scripts/buildsystems/vcpkg.cmake:843 (_find_package)
  CMakeLists.txt:12 (find_package)


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**Additional context**

Add any other context about the problem here, such as what you have already tried to resolve the issue.

--- a/ports/gamenetworkingsockets/fix-depend-protobuf.patch
+++ b/ports/gamenetworkingsockets/fix-depend-protobuf.patch
@@ -1,0 +1,12 @@
+diff --git a/cmake/GameNetworkingSocketsConfig.cmake.in b/cmake/GameNetworkingSocketsConfig.cmake.in
+index 1a27988..5076a6e 100644
+--- a/cmake/GameNetworkingSocketsConfig.cmake.in
++++ b/cmake/GameNetworkingSocketsConfig.cmake.in
+@@ -3,6 +3,7 @@
+ include(CMakeFindDependencyMacro)
+ 
+ find_dependency(Threads)
++find_dependency(Protobuf)
+ 
+ if(@USE_CRYPTO@ STREQUAL "OpenSSL")
+     find_dependency(OpenSSL)

--- a/ports/gamenetworkingsockets/portfile.cmake
+++ b/ports/gamenetworkingsockets/portfile.cmake
@@ -6,6 +6,8 @@ vcpkg_from_github(
     REF 505c697d0abef5da2ff3be35aa4ea3687597c3e9 # v1.4.1
     SHA512 3e4b4da138f2b356169e6504aa899c9eca4fba5b5fcaed2a0ae8a2f5828976dd00af9f3262c75bd6d820300da87ebe32da152fecddc278a651f3b33eb59142df
     HEAD_REF master
+    PATCHES
+        fix-depend-protobuf.patch
 )
 
 set(CRYPTO_BACKEND OpenSSL)
@@ -24,12 +26,6 @@ vcpkg_cmake_install()
 
 vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/GameNetworkingSockets")
 vcpkg_fixup_pkgconfig()
-
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/${PORT}/GameNetworkingSocketsConfig.cmake"
-[[find_dependency(Threads)]]
-[[find_dependency(Threads)
-find_dependency(protobuf CONFIG REQUIRED)]]
-)
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")

--- a/ports/gamenetworkingsockets/portfile.cmake
+++ b/ports/gamenetworkingsockets/portfile.cmake
@@ -21,12 +21,18 @@ vcpkg_cmake_configure(
 )
 
 vcpkg_cmake_install()
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
 
 vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/GameNetworkingSockets")
 vcpkg_fixup_pkgconfig()
+
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/${PORT}/GameNetworkingSocketsConfig.cmake"
+[[find_dependency(Threads)]]
+[[find_dependency(Threads)
+find_dependency(protobuf CONFIG REQUIRED)]]
+)
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
 vcpkg_copy_pdbs()
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/gamenetworkingsockets/vcpkg.json
+++ b/ports/gamenetworkingsockets/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "gamenetworkingsockets",
-  "version-semver": "1.4.1",
+  "version": "1.4.1",
+  "port-version": 1,
   "description": "GameNetworkingSockets is a basic transport layer for games.",
   "homepage": "https://github.com/ValveSoftware/GameNetworkingSockets",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2542,7 +2542,7 @@
     },
     "gamenetworkingsockets": {
       "baseline": "1.4.1",
-      "port-version": 0
+      "port-version": 1
     },
     "gamma": {
       "baseline": "gamma-2018-01-27",

--- a/versions/g-/gamenetworkingsockets.json
+++ b/versions/g-/gamenetworkingsockets.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "a91f7999c277c68b2034ad357b99cb7acf7b69ce",
+      "git-tree": "75eaecf85e526c7de535e77ac6da7025c8e4020f",
       "version": "1.4.1",
       "port-version": 1
     },

--- a/versions/g-/gamenetworkingsockets.json
+++ b/versions/g-/gamenetworkingsockets.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a91f7999c277c68b2034ad357b99cb7acf7b69ce",
+      "version": "1.4.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "94a9a726c8c2caf64acec0988bd9ad58f57ab5df",
       "version-semver": "1.4.1",
       "port-version": 0

